### PR TITLE
Fixes for YARD docs

### PR DESCRIPTION
**Pull request by katafrakt**
This fixes some things in YARD docs. More of a housekeeping than storytelling - I focused on minor things like:
* Making sure links in @see work
* Transforming all descriptions to use 3rd person (finalize -> finalizes etc.)
* Fixing wording in few places
* Deleting some outdated examples and links
* Fixing formatting in few cases

### CODE REVIEW

File: lib/hanami/controller.rb [hunk 1]
```diff
@@ -13,19 +13,6 @@ module Hanami
...
-  #
-  #   module Articles
-  #     class Index < Hanami::Action
-  #       # ...
-  #     end
-  #
-  #     class Show < Hanami::Action
-  #       # ...
-  #     end
-  #   end
```
Comment (katafrakt) [inline lib/hanami/controller.rb:28]: This example was weird - as it does not show anything about `Hanami::Controller`, I decided to delete it

File: lib/hanami/http/status.rb [hunk 2]
```diff
@@ -27,19 +27,24 @@ class Status
       # @example
       #   require 'hanami/http/status'
       #
-      #   Hanami::Http::Status.for_code(418) # => [418, "I'm a teapot"]
+      #   Hanami::Http::Status.for_code(409) # => [409, "Conflict"]
```
Comment (katafrakt) [inline lib/hanami/http/status.rb:30]: Actually 418 is probably the only status that is not supported (due to lack of support in `Rack::Utils`)

File: lib/hanami/http/status.rb [hunk 3]
```diff
@@ -27,19 +27,24 @@ class Status
...
       #
       # @param code [Integer] a valid HTTP code
       #
       # @return [String] a message for the given status code
       #
       # @since 0.3.2
       # @api private
+      #
+      # @example
+      #   require 'hanami/http/status'
```
Comment (jodosha) [inline lib/hanami/http/status.rb:45]:
```suggestion
      #   require "hanami/http/status"
```

**Review by jodosha — APPROVED** [review]

**Comment by jodosha** [comment]
@katafrakt Thank you!

File: lib/hanami/http/status.rb (hunk 3 above)
Comment (katafrakt) [reply lib/hanami/http/status.rb]: Oh, right, I haven't checked for single quotes in code examples. Probably there's more of these (for example few lines above, from which I copied this example). Should I review it as well before this is merged?

**Comment by katafrakt** [comment]
@jodosha I rebased this onto `main` to resolve conflicts. However, you assign it to me, but I'm not sure what more I should do with this PR ;)

**Comment by jodosha** [comment]
@katafrakt Thank you, closing in favor of https://github.com/hanami/controller/pull/401

**Pull request closed without merging**